runtime/drivers/duckdb/duckdb.go

```diff
@@ -37,46 +45,111 @@ func (d driver) Open(dsn string) (drivers.Connection, error) {
 		"SET max_expression_depth TO 250",
 	}
 
+	metaConn, err := db.Connx(context.Background())
+	if err != nil {
+		return nil, err
+	}
 	for _, qry := range bootQueries {
-		_, err = db.Exec(qry)
+		_, err = metaConn.ExecContext(context.Background(), qry)
 		if err != nil {
 			return nil, err
 		}
 	}
 
-	conn := &connection{db: db}
-	conn.worker = priorityworker.New(conn.executeQuery)
+	c := &connection{
+		db:       db,
+		metaConn: metaConn,
+		pool:     make(chan *sqlx.Conn, cfg.PoolSize),
+		sem:      priorityqueue.NewSemaphore(cfg.PoolSize),
+		closed:   false,
+	}
+
+	for i := 0; i < cfg.PoolSize; i++ {
+		conn, err := db.Connx(context.Background())
+		if err != nil {
+			return nil, err
+		}
+
+		for _, qry := range bootQueries {
+			_, err = conn.ExecContext(context.Background(), qry)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		c.pool <- conn
+	}
 
-	return conn, nil
+	return c, nil
 }
 
 type connection struct {
-	db     *sqlx.DB
-	worker *priorityworker.PriorityWorker[*job]
+	db         *sqlx.DB
+	metaConn   *sqlx.Conn
+	pool       chan *sqlx.Conn
+	sem        *priorityqueue.Semaphore
+	closed     bool
+	closeMutex sync.Mutex
 }
 
 // Close implements drivers.Connection.
 func (c *connection) Close() error {
-	c.worker.Stop()
+	c.closeMutex.Lock()
+	c.closed = true
+	close(c.pool)
+	c.closeMutex.Unlock()
+	var firstErr error
+	for conn := range c.pool {
+		err := conn.Close()
+		if err != nil && firstErr == nil {
+			firstErr = err
+		}
+	}
+	if firstErr != nil {
+		return firstErr
+	}
 	return c.db.Close()
 }
 
-// Registry implements drivers.Connection.
+// RegistryStore Registry implements drivers.Connection.
 func (c *connection) RegistryStore() (drivers.RegistryStore, bool) {
 	return nil, false
 }
 
-// Catalog implements drivers.Connection.
+// CatalogStore Catalog implements drivers.Connection.
 func (c *connection) CatalogStore() (drivers.CatalogStore, bool) {
 	return c, true
 }
 
-// Repo implements drivers.Connection.
+// RepoStore Repo implements drivers.Connection.
 func (c *connection) RepoStore() (drivers.RepoStore, bool) {
 	return nil, false
 }
 
-// OLAP implements drivers.Connection.
+// OLAPStore OLAP implements drivers.Connection.
 func (c *connection) OLAPStore() (drivers.OLAPStore, bool) {
 	return c, true
 }
+
+// getConn gets a connection from the pool.
+// It returns a function that puts the connection back in the pool if applicable.
+func (c *connection) getConn(ctx context.Context) (conn *sqlx.Conn, release func(), err error) {
+	// Try to get conn from context
+	conn = connFromContext(ctx)
+	if conn != nil {
+		return conn, func() {}, nil
+	}
+
+	conn, ok := <-c.pool
+	if !ok {
+		return nil, nil, drivers.ErrClosed
+	}
+	fn := func() {
+		c.closeMutex.Lock()
+		if !c.closed {
+			c.pool <- conn
+		}
+		c.closeMutex.Unlock()
+	}
+	return conn, fn, nil
```

Comment:
With the DuckDB conn init PR, we can use a built-in connection pool, so we just need to do something like:
```go
conn, err := c.db.Conn()
if err != nil {
  return nil, nil, err
}
return conn, conn.Close, nil
```

Reply:
done